knnga_util.py

```diff
@@ -0,0 +1,593 @@
+# Copyright (C) 2020 Juliette Regimbal
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import unicode_literals
+from gamera import knnga
+
+import json
+
+
+class SerializableSelection():
+    """
+    Extension of gamera.knnga.GASelection that provides
+    access to the selection and parameters used.
+    """
+
+    def __init__(self):
+        self.method = None
+        self.parameters = {}
+        self.selection = knnga.GASelection()
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            a = self.__dict__.copy()
+            del a['selection']
+            b = self.__dict__.copy()
+            del b['selection']
+            return a == b
```

Comment:
Doesn't this compare the object to itself? Shouldn't it be something like:

```python
if isinstance(other, self.__class__):
    a = self.__dict__.copy()
    del a['selection']
    b = other.__dict__.copy()
    del b['selection']
    return a == b
```

Reply:
Oops thanks I missed this